Content.Server/Clothing/MaskSystem.cs

```diff
@@ -0,0 +1,104 @@
+using Content.Shared.Actions;
+using Content.Shared.Toggleable;
+using Content.Shared.Inventory;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Item;
+using Content.Server.Atmos.Components;
+using Content.Server.Body.Components;
+using Content.Server.Clothing.Components;
+using Content.Server.Disease.Components;
+using Content.Server.Nutrition.EntitySystems;
+using Content.Server.Popups;
+using Robust.Shared.Player;
+
+namespace Content.Server.Clothing
+{
+    public sealed class MaskSystem : EntitySystem
+    {
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly InventorySystem _inventorySystem = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<MaskComponent, ToggleMaskEvent>(OnToggleMask);
+            SubscribeLocalEvent<MaskComponent, GetItemActionsEvent>(OnGetActions);
+            SubscribeLocalEvent<MaskComponent, GotUnequippedEvent>(OnGotUnequipped);
+        }
+
+        private void OnGetActions(EntityUid uid, MaskComponent component, GetItemActionsEvent args)
+        {
+            if (component.ToggleAction != null)
+                args.Actions.Add(component.ToggleAction);
+        }
+
+        private void OnToggleMask(EntityUid uid, MaskComponent mask, ToggleMaskEvent args)
+        {
+            if (mask.ToggleAction == null)
+                return;
+
+            // return if we equipped, or failed to equip due to something else in mask slot
+            if (_inventorySystem.TryEquip(args.Performer, mask.Owner, "mask") ||
+                (_inventorySystem.TryGetSlotEntity(args.Performer, "mask", out var existing) && !mask.Owner.Equals(existing)))
+                return;
```

Comment:
all these checks other than the one checking if the mask is equipped in the mask slot make little sense imo. also change the if statement to an or so it also fails if it cant find any entity in the mask slot
```suggestion
            if (!_inventorySystem.TryGetSlotEntity(args.Performer, "mask", out var existing) || !mask.Owner.Equals(existing))
                return;
```

Reply:
I was adding the functionality of equipping it on toggle if you didn't have it equipped yet. However, your change makes the logic simpler and would avoid someone accidentally double clicking and equipping then pulling down without realizing it, so that sounds good to me.